catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/util/MetacardServices.java

```diff
@@ -0,0 +1,104 @@
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableList;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+
+/**
+ * Support and bulk operations for {@link Metacard}s. Contains state relevant to the provided
+ * services.
+ */
+public class MetacardServices {
+
+    private final List<MetacardType> systemMetacardTypes;
+
+    /**
+     * Initializes the services with an empty list of system {@link MetacardType}s.
+     */
+    public MetacardServices() {
+        this.systemMetacardTypes = new ArrayList<>();
+    }
+
+    /**
+     * Initializes the services with the provided list of {@link MetacardType}s.
+     *
+     * @param systemMetacardTypes The list of metacard types to use.
+     */
+    public MetacardServices(List<MetacardType> systemMetacardTypes) {
+        this.systemMetacardTypes = systemMetacardTypes;
+    }
+
+    /**
+     * Iterates over each {@link Metacard} in the {@param metacards} collection and applies each
+     * attribute in the {@param attributeMap} only if the attribute is not already set on the
+     * {@link Metacard}. That is, the attribute must be {@code null} or not in the {@link Metacard}'s
+     * map.
+     *
+     * @param metacards        The list of metacards to attempt to add the attributes to.
+     * @param attributeMap     The map of attributes to attempt to add. Attributes already set on any
+     *                         given {@link Metacard} will not be changed. For multi-valued attributes,
+     *                         the value string should separate different entities with commas.
+     * @param attributeFactory The factory to use to create attributes.
+     */
+    public void setAttributesIfAbsent(List<Metacard> metacards, Map<String, String> attributeMap,
+            AttributeFactory attributeFactory) {
+
+        if (metacards.isEmpty() || attributeMap.isEmpty()) {
+            return;
+        }
+
+        List<MetacardType> systemMetacardTypesCopy = ImmutableList.copyOf(systemMetacardTypes);
+
+        Map<String, AttributeDescriptor> systemAndMetacardDescriptors = Stream.concat(
+                systemMetacardTypesCopy.stream()
+                        .map(MetacardType::getAttributeDescriptors)
+                        .flatMap(Set::stream)
+                        // ** Does our API publish null descriptors? **
+                        .filter(Objects::nonNull),
+                metacards.stream()
+                        .map(Metacard::getMetacardType)
+                        .map(MetacardType::getAttributeDescriptors)
+                        .flatMap(Set::stream)
+                        // ** Does our API allow null descriptors on metacards? **
+                        .filter(Objects::nonNull))
+
+                .collect(Collectors.toMap(AttributeDescriptor::getName,
+                        Function.identity(),
+                        (oldValue, newValue) -> oldValue));
```

Comment:
This is the correct behavior we must ensure. First in wins (because legacy that's how it behaved)